godump.go

```diff
@@ -966,3 +958,56 @@ func contains(candidates []reflect.Kind, target reflect.Kind) bool {
 
 	return false
 }
+
+func isComplexValue(v reflect.Value) bool {
+	_, ok := complexBaseKind(v)
+	return ok
+}
+
+func complexBaseKind(v reflect.Value) (reflect.Kind, bool) {
+	if !v.IsValid() {
+		return 0, false
+	}
+
+	for v.Kind() == reflect.Interface {
+		if v.IsNil() {
+			return 0, false
+		}
+		v = v.Elem()
+	}
+
+	for v.Kind() == reflect.Ptr {
+		if v.IsNil() {
+			return 0, false
+		}
+		v = v.Elem()
+	}
```

Comment:
Are you sure it shouldn't be something like this ?


```suggestion
	for {
		switch v.Kind() {
			case reflect.Interface:
				if v.IsNil() {
					return 0, false
				}
				v = v.Elem()
			case reflect.Ptr:
				if v.IsNil() {
					return 0, false
				}
				v = v.Elem()
			default:
				break
	}
```

Reply:
`complexBaseKind` now loops until the value peels away all interfaces and pointers (handling nested combinations like *interface{}) before deciding whether it’s a struct/map/slice/array, so we don’t miss a case where a pointer unwrap yields another interface